build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```
@@ -0,0 +1,78 @@
+import java.nio.charset.StandardCharsets
+import java.util.Optional
+import java.util.jar.JarFile
+import java.util.stream.Stream
+import java.util.stream.Collectors
+import groovy.transform.Field
+
+/**
+ * This Gradle script captures Predictive Test Selection and Test Distribution compatibility for each Test task, 
+ * adding a flag as custom value.
+ * This should be applied to the root project since it configures all projects:
+ * <code> apply from: file('gradle-test-pts-support.gradle') </code>
+ */
+
+def buildScanApi = project.extensions.findByName('buildScan')
+if (!buildScanApi) {
+    return
+}
+
+@Field
+final def supportedEngines = [
+    'org.junit.support.testng.engine.TestNGTestEngine' : 'testng',
+    'org.junit.jupiter.engine.JupiterTestEngine' : 'junit-jupiter', 
+    'org.junit.vintage.engine.VintageTestEngine' : 'junit-vintage', 
+    'org.spockframework.runtime.SpockEngine' : 'spock',
+    'net.jqwik.engine.JqwikTestEngine' : 'jqwik',
+    'io.cucumber.junit.platform.engine.CucumberTestEngine' : 'cucumber',
+    'org.spekframework.spek2.junit.SpekTestEngine' : 'spek2',
+    'com.tngtech.archunit.junit.ArchUnitTestEngine' : 'archunit'
+    ]
+
+allprojects {
+    tasks.withType(Test).configureEach { t ->
+        doFirst {
+            if (t.getTestFramework().getClass().getName() == 'org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework') {
+                def engines = testEngines(t) 
+                buildScanApi.value("${t.identityPath}#engines", "${engines}")
+                if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
+                    buildScanApi.value("${t.identityPath}#pts", 'SUPPORTED')
+                } else {
+                    buildScanApi.value("${t.identityPath}#pts", 'ENGINES_NOT_ALL_SUPPORTED')
+                }
+            } else {
+                buildScanApi.value("${t.identityPath}#pts", 'NO_JUNIT_PLATFORM')
+            }
+        }
+    }
+}
+
+Set<String> testEngines(Task t) {
```

The plain `Task` doesn't have a `classpath`
```suggestion
Set<String> testEngines(Test t) {
```